kubera/dmaas/credentials/minio_test.go

```diff
@@ -0,0 +1,44 @@
+package minio
+
+import (
+	"testing"
+
+	auth "github.com/mayadata-io/kubera-api-testing/kubera/authentication"
+	"github.com/mayadata-io/kubera-api-testing/kubera/connection"
+)
+
+func TestMinioCredentialUpdate(t *testing.T) {
+	credentialTitle := "minio-cred-demo"
+	cloudProvider := "minio"
+	conn := connection.Get()
+	fetcher := auth.NewTokenFetcher(auth.TokenFetchingConfig{
+		Connection: conn,
+	})
+	tkn, err := fetcher.Fetch()
+	if err != nil {
+		t.Fatalf("Expected no error got=%s: \nFetcher=%s: \nToken=%s", err, fetcher, tkn)
+	}
+	// Fetch group id
+	groupID, err := getGroupID(tkn)
+	if err != nil {
+		t.Fatalf("Error while fetching group ID=%s: ", err)
+	}
+
+	// fetch project id belongs to above group
+	projectID, err := getProjectID(tkn)
+	if err != nil {
+		t.Fatalf("Error while fetching project ID =%s: ", err)
+	}
+	/* Director supports a fixed set of backup providers.
+	Backup providers are those that provide object storage to backup the data. */
+	backupProviderID, err := getBackupProviderID(tkn, cloudProvider)
+	if err != nil {
+		t.Fatalf("Error while fetching backup provider ID =%s: ", err)
+	}
+
+	// add cloud credential and record error in any
+	error := addMinioCloudCredential(tkn, groupID, projectID, backupProviderID, credentialTitle)
+	if error != nil {
+		t.Fatalf("Error while adding %s cloud credential=%s: ", cloudProvider, err)
+	}
+}
```

Comment:
On the whole, we need to ensure testing adding minio or aws or cloudian etc credentials is seamless.
IMO minio, aws, cloudian etc are a part of various combinations that should be tested within _test.go file.

Reply:
```go
// ---
 
 type KuberaGroup struct {
   // fields that are returned as payload
 }
 
 type KuberaGroupFetching struct {
   // fields that are needed to fetch kubera group
 }
 
 func NewKuberaGroupFetcher() *KuberaGroupFetching {
    return &KuberaGroupFetching{}
 }
 
 func (f *KuberaGroupFetching) Fecth() (KuberaGroup, error) {
  // rest api call 
 }
 
//  ---
 
 type KuberaProject struct {
   // fields that are returned as payload
 }
 
 type KuberaProjectFetching struct {}
 
 func NewKuberaProjectFetcher() *KuberaProjectFetching{
   return &KuberaProjectFetching{}
 }
 
  func (f *KuberaProjectFetching) Fetch() (KuberaProject, error) {
  // rest api call 
 }
 
//  ---
 
 type BackupProviderList struct {
   // fields that are returned as payload
 }
 
 type BackupProvider struct {
  // fields that represent payloads && all kinds of backup providers
  // including AWS, Minio
 }
 
 type CloudianBackupProvider struct {
   BackupProvider
   // fields that are specific to cloudian
 }
 
 type BackupProviderListFetching struct {}
 
 func NewBackupProviderListFetcher() *BackupProviderListFetching{
   return &BackupProviderListFetching{}
 }

 func (l *BackupProviderListFetching) FetchAll() (BackupProviderList, error) {
  // rest api invocation
 }
 
 func (l *BackupProviderListFetching) FetchMinioProvider() (BackupProvider, error) {
    list, err := l.FetchAll()
    // extract the minio specific BackupProvider & return
 }
 
// ---
 
 type Credentials struct {
   // fields should represent the payload
 }
 
 type CredentialsAdding struct {
   // fields should represent the ingredients requried to add a credential
 }
 
 func NewCredentialsAdder() *CredentialsAdding {
   return &CredentialsAdding{}
 }
 
 func (a *CredentialsAdding) Add() (Credentials, error) {
   // rest api invocations
   // -- invoke group api
   // -- invoke project api
   // -- invoke backup provider list api 
   // -- invoke add credentials api
 }
 ```